src/Plugin/RulesAction/EntityCreateDeriver.php

```diff
@@ -92,6 +92,7 @@ public function getDerivativeDefinitions($base_plugin_definition) {
         $required = ($field_name == $bundle_key);
         $multiple = ($definition->getCardinality() === 1) ? FALSE : TRUE;
         $this->derivatives["entity:$entity_type_id"]['context'][$field_name] = ContextDefinition::create($definition->getType())
+          ->setAssignmentRestriction(ContextDefinition::ASSIGNMENT_RESTRICTION_INPUT)
```

Comment:
we should probably add test coverage for this. Add a second base field in the setUp() method of the test which is required. Add a third base field which is not required. Check that the generated derivative has 2 contexts, one for the bundle and one for the required base field. Check that the assignment restriction is only set on the bundle, not on the second field.